rpc/core/blocks.go

```diff
@@ -131,6 +134,73 @@ func Commit(ctx *rpctypes.Context, heightPtr *int64) (*ctypes.ResultCommit, erro
 	return ctypes.NewResultCommit(&header, commit, true), nil
 }
 
+// DataCommitment collects the data roots over a provided ordered range of blocks,
+// and then creates a new Merkle root of those data roots.
+func DataCommitment(ctx *rpctypes.Context, beginBlock uint64, endBlock uint64) (*ctypes.ResultDataCommitment, error) {
+	err := validateDataCommitmentRange(beginBlock, endBlock)
+	if err != nil {
+		return nil, err
+	}
+	heights := generateHeightsList(beginBlock, endBlock)
+	blockResults := fetchBlocks(heights, len(heights), 0)
+	root := hashDataRoots(blockResults)
+	// Create data commitment
+	return &ctypes.ResultDataCommitment{DataCommitment: root}, nil
+}
+
+// generateHeightsList takes a begin and end block, then generates a list of heights
+// containing the elements of the range [beginBlock, endBlock].
+func generateHeightsList(beginBlock uint64, endBlock uint64) []int64 {
+	heights := make([]int64, endBlock-beginBlock+1)
+	for i := beginBlock; i <= endBlock; i++ {
+		heights[i-beginBlock] = int64(i)
+	}
+	return heights
+}
+
+// validateDataCommitmentRange runs basic checks on the asc sorted list of heights
+// that will be used subsequently in generating data commitments over the defined set of heights.
+func validateDataCommitmentRange(beginBlock uint64, endBlock uint64) error {
+	heightsRange := endBlock - beginBlock + 1
+	if heightsRange > uint64(consts.DataCommitmentBlocksLimit) {
+		return fmt.Errorf("the query exceeds the limit of allowed blocks %d", consts.DataCommitmentBlocksLimit)
+	}
+	if heightsRange == 0 {
+		return fmt.Errorf("cannot create the data commitments for an empty set of blocks")
+	}
+	if beginBlock > endBlock {
+		return fmt.Errorf("end block is smaller than begin block")
+	}
```

Comment:
```suggestion
	if beginBlock > endBlock {
		return fmt.Errorf("end block is smaller than begin block")
	}
	heightsRange := endBlock - beginBlock + 1 // The check above guarantees this does not wrap
	if heightsRange > uint64(consts.DataCommitmentBlocksLimit) {
		return fmt.Errorf("the query exceeds the limit of allowed blocks %d", consts.DataCommitmentBlocksLimit)
	}
	if heightsRange == 0 {
		return fmt.Errorf("cannot create the data commitments for an empty set of blocks")
	}
```

The fact that `beginBlock` and `endBlock` are unsigned integers have important implications in these checks, since the substraction MAY wrap. Therefore, the order in which these checks are made may produce misleading errors to the user. I have rearranged the check in my suggestion to generate accurate error as much as possible.